File: app/src/main/java/io/github/ovso/righttoknow/framework/BaseActivity.java
package io.github.ovso.righttoknow.framework;

import android.os.Build;
import android.os.Bundle;
import android.view.ViewGroup;
import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.core.content.ContextCompat;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import dagger.android.support.DaggerAppCompatActivity;
import io.github.ovso.righttoknow.R;
import io.github.ovso.righttoknow.framework.ad.MyAdView;

public abstract class BaseActivity extends DaggerAppCompatActivity {

  private Unbinder unbinder;

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(getLayoutResId());
    unbinder = ButterKnife.bind(this);
    final Toolbar toolbar = findViewById(R.id.toolbar);
    if (toolbar != null) {
      setSupportActionBar(toolbar);
    }

    setNavigationBarColor();
    showAd();
  }

  protected void showAd() {
    ViewGroup adContainer = findViewById(R.id.ad_container);
    if (adContainer != null) {
      adContainer.addView(MyAdView.getAdmobAdView(getApplicationContext()));
    }
  }

  private void setNavigationBarColor() {
    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
      getWindow().setNavigationBarColor(ContextCompat.getColor(this, R.color.colorPrimary));
    }
  }

  @LayoutRes protected abstract int getLayoutResId();

  @Override protected void onDestroy() {
    super.onDestroy();
    unbinder.unbind();
  }

  protected Toolbar getToolbar() {
    return findViewById(R.id.toolbar);
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/framework/BaseActivity.kt
package io.github.ovso.righttoknow.framework

import android.os.Build
import android.os.Bundle
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.content.ContextCompat
import io.github.ovso.righttoknow.R
import io.github.ovso.righttoknow.framework.ad.MyAdView

abstract class BaseActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    val toolbar = findViewById<Toolbar>(R.id.toolbar)
    toolbar?.let { setSupportActionBar(it) }
    setNavigationBarColor()
  }

  protected fun showAds() {
    val adContainer = findViewById<ViewGroup>(R.id.ad_container)
    adContainer?.addView(MyAdView.getAdmobAdView(applicationContext))
  }

  private fun setNavigationBarColor() {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
      window.navigationBarColor = ContextCompat.getColor(this, R.color.colorPrimary)
    }
  }

  protected val toolbar: Toolbar
    protected get() = findViewById(R.id.toolbar)
}
